Send signed transaction to deleted chats

cammellos opened the pull request:
fixes #3624 

### Summary:

When sending a transaction using sign later and the chat has been deleted, an error was displayed. 

I have changed so that we still add the message to the deleted chat.

I have also removed the old `upsert-chat` and renamed `update-chat` to `upsert-chat` as that is a more appropriate name.

Fixes also the error message displayed when sending messages to console.

status: ready

cammellos committed:
Send signed transaction to deleted chats

janherich commented:
@cammellos I'm afraid we are still storing "deleted" (soft delete, we so it's still in realm, just `:ist-active?` is set to false) in realm and sometimes we need to read them, typical example is when you delete some chat (doesn't matter if it's 1-1 or group/public) and new messages for such chat arrives - in that case, we look if the chat id of the message is in `:deleted-chats` set and when yes, we don't create new chat entity, we rather load deleted chat entity from db and set `:is-active?` to true (so it's "resurrected").

Maybe we don't need do do that (completely wipe chats from `app-db`, and re-fetch them from realm when necessary) and when deleting chat, we should just set flag `is-active?` in `app-db` and `realm`, then, no realm re-fetching would be necessary.

cammellos commented:
@janherich You are correct, I did not see that we only delete chat in debug mode, I will have a go again :(

cammellos commented:
@janherich Changed the code so we don't use `deleted-chats` anymore, we just set  `is-active` to false. 

I have also removed `add-chat` as mostly a duplication of `upsert-chat` and fixed a bug with leaving/deleting a public chat ( we were not re-adding the filter, which meant that the user was not able to communicate properly.

Current behaviour is:

1) On delete a public/group/1to1 chat -> On receiving of new message it will reappear.
2) On leave a public-chat/group-chat -> No further message will be received ( group chat messages sent to the pk are still problematic, haven't fixed that, but will take care in another pr)

dmitryn commented on src/status_im/chat/models.cljs:
```diff
@@ -25,71 +25,39 @@
+  [topic cofx]
+  (upsert-chat {:chat-id topic
+                :is-active true
+                :name             topic
```
values align

dmitryn commented on src/status_im/chat/models.cljs:
```diff
@@ -25,71 +25,39 @@
-  (update-chat (assoc chat :timestamp (:now cofx)) cofx))
+  [chat-id chat-name admin participants cofx]
+  (upsert-chat {:chat-id chat-id
+                :name chat-name
```
same, map values align

asemiankevich commented:
@cammellos i think the deleted chat should be shown again (on the Sender side) in this case, but not sure how it is treated programmatically, so it may be a bit tricky? Could you please elaborate? Note, that everything looks good on the recipient side (chat is shown after i deleted it - sent transaction from unsigned)

It could be another ticket, i am pretty sure the same is in develop despite the error that is fixed in this PR and i confirm that

cammellos commented:
@asemiankevich maybe best to track in a different issue, as this PR has already some significant changes, if you don't mind.
cheers

asemiankevich commented:
@cammellos sure!
